lib/std/fs.zig

```diff
@@ -2240,12 +2240,22 @@ pub const Dir = struct {
 
     pub const StatFileError = File.OpenError || StatError;
 
-    // TODO: improve this to use the fstatat syscall instead of making 2 syscalls here.
-    pub fn statFile(self: Dir, sub_path: []const u8) StatFileError!File.Stat {
-        var file = try self.openFile(sub_path, .{});
-        defer file.close();
+    // TODO: simplify this once windows get a proper fstatat implementation.
+    pub fn statFile(self: Dir, sub_path: []const u8) if (builtin.os.tag == .windows) StatFileError!Stat else os.FStatAtError!Stat {
```

Comment:
For simplicity, the current policy of the standard library is to make the error sets the same regardless of operating system. You can see this in the other functions in this file if you need an example.